agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java

```diff
@@ -0,0 +1,337 @@
+      List<Class<?>> implementingClasses = implementingClassesCache.get(type);
+      if (implementingClasses == null) {
+        ClassGraph classGraph =
+            new ClassGraph().enableClassInfo().enableInterClassDependencies().rejectPackages(
+                "jaz.*");
+        if (!isTest()) {
+          classGraph.rejectPackages("com.code_intelligence.jazzer.*");
+        }
+        try (ScanResult result = classGraph.scan()) {
+          ClassInfoList children =
+              type.isInterface() ? result.getClassesImplementing(type) : result.getSubclasses(type);
+          implementingClasses =
+              children.getStandardClasses().filter(cls -> !cls.isAbstract()).loadClasses();
+          implementingClassesCache.put(type, implementingClasses);
+        }
+      }
+      if (implementingClasses.isEmpty()) {
+        if (isDebug()) {
+          throw new AutofuzzConstructionException(String.format(
+              "Could not find classes implementing %s on the classpath", type.getName()));
+        } else {
+          throw new AutofuzzConstructionException();
+        }
+      }
+      return consume(data, data.pickValue(implementingClasses));
+    } else if (type.getConstructors().length > 0) {
+      Constructor<?> constructor = data.pickValue(type.getConstructors());
+      Object obj = autofuzz(data, constructor);
+      if (constructor.getParameterCount() == 0) {
+        List<Method> potentialSetters = getPotentialSetters(type);
+        if (!potentialSetters.isEmpty()) {
+          List<Method> pickedSetters =
+              data.pickValues(potentialSetters, data.consumeInt(0, potentialSetters.size()));
+          for (Method setter : pickedSetters) {
+            autofuzz(data, setter, obj);
+          }
+        }
+      }
+      return obj;
+    }
+    // We are out of more or less canonical ways to construct an instance of this class and have to
+    // resort to more heuristic approaches.
+
+    // First, try to find nested classes with names ending in Builder and call a subset of their
+    // chaining methods.
+    List<Class<?>> nestedBuilderClasses = getNestedBuilderClasses(type);
+    if (!nestedBuilderClasses.isEmpty()) {
+      Class<?> pickedBuilder = data.pickValue(nestedBuilderClasses);
+
+      List<Method> cascadingBuilderMethods = Arrays.stream(pickedBuilder.getMethods())
+                                                 .filter(m -> m.getReturnType() == pickedBuilder)
+                                                 .collect(Collectors.toList());
+
+      List<Method> originalObjectCreationMethods = getOriginalObjectCreationMethods(pickedBuilder);
+
+      int pickedMethodsNumber = data.consumeInt(0, cascadingBuilderMethods.size());
+      List<Method> pickedMethods = data.pickValues(cascadingBuilderMethods, pickedMethodsNumber);
+
+      Method builderMethod = data.pickValue(originalObjectCreationMethods);
+
+      Object builderObj = autofuzz(data, data.pickValue(pickedBuilder.getConstructors()));
```

Comment:
Should we check here for returned 'null' objects?

Reply:
See above, since this calls through to `Constructor.newInstance`, I believe it can never return `null`.